core/domain/story_domain.py

```diff
@@ -104,85 +108,132 @@ class StoryChange(change_domain.BaseChange):
         'user_id_attribute_names': [],
-        'allowed_values': {'property_name': STORY_PROPERTIES}
+        'allowed_values': {'property_name': STORY_PROPERTIES},
+        'deprecated_values': {}
     }, {
         'name': CMD_UPDATE_STORY_NODE_PROPERTY,
         'required_attribute_names': [
             'node_id', 'property_name', 'new_value', 'old_value'],
         'optional_attribute_names': [],
         'user_id_attribute_names': [],
-        'allowed_values': {'property_name': STORY_NODE_PROPERTIES}
+        'allowed_values': {'property_name': STORY_NODE_PROPERTIES},
+        'deprecated_values': {}
     }, {
         'name': CMD_UPDATE_STORY_CONTENTS_PROPERTY,
         'required_attribute_names': ['property_name', 'new_value', 'old_value'],
         'optional_attribute_names': [],
         'user_id_attribute_names': [],
-        'allowed_values': {'property_name': STORY_CONTENTS_PROPERTIES}
+        'allowed_values': {'property_name': STORY_CONTENTS_PROPERTIES},
+        'deprecated_values': {}
     }, {
         'name': CMD_ADD_STORY_NODE,
         'required_attribute_names': ['node_id', 'title'],
         'optional_attribute_names': [],
-        'user_id_attribute_names': []
+        'user_id_attribute_names': [],
+        'allowed_values': {},
+        'deprecated_values': {}
     }, {
         'name': CMD_DELETE_STORY_NODE,
         'required_attribute_names': ['node_id'],
         'optional_attribute_names': [],
-        'user_id_attribute_names': []
+        'user_id_attribute_names': [],
+        'allowed_values': {},
+        'deprecated_values': {}
     }, {
         'name': CMD_UPDATE_STORY_NODE_OUTLINE_STATUS,
         'required_attribute_names': ['node_id', 'old_value', 'new_value'],
         'optional_attribute_names': [],
-        'user_id_attribute_names': []
+        'user_id_attribute_names': [],
+        'allowed_values': {},
+        'deprecated_values': {}
     }, {
         'name': CMD_CREATE_NEW,
         'required_attribute_names': ['title'],
         'optional_attribute_names': [],
-        'user_id_attribute_names': []
+        'user_id_attribute_names': [],
+        'allowed_values': {},
+        'deprecated_values': {}
     }, {
         'name': CMD_MIGRATE_SCHEMA_TO_LATEST_VERSION,
         'required_attribute_names': ['from_version', 'to_version'],
         'optional_attribute_names': [],
-        'user_id_attribute_names': []
+        'user_id_attribute_names': [],
+        'allowed_values': {},
+        'deprecated_values': {}
     }]
 
 
+class StoryNodeDict(TypedDict):
+    """Dictionary representing the StoryNode object."""
+
+    id: str
+    title: str
+    description: str
+    thumbnail_filename: Optional[str]
+    thumbnail_bg_color: Optional[str]
+    thumbnail_size_in_bytes: Optional[int]
+    destination_node_ids: List[str]
+    acquired_skill_ids: List[str]
+    prerequisite_skill_ids: List[str]
+    outline: str
+    outline_is_finalized: bool
+    exploration_id: Optional[str]
```

Comment:
Does this need to be `Optional`, I think we have discussed that before?

Reply:
I think we only discussed `app_feedback_report_domain.py`'s story_id. However, if we remove optional from here then mypy throws more than 20+ errors from test files because there we are using None. Also, while creating the default_story_node we are providing none value.

Reply:
Hmm, test files are always fixable, can you investigate why we are providing `None` in `default_story_node`?

Reply:
While creating `default_story_node` we are providing only node_id and title. Also, `exploration_id` can be None when the story creator just created a story with a basic storyline ( As per the doc string ). 

If we want `exploration_id` to be str type we need to change the `default_story_node`'s exploration_id from None to '' empty string. But I'm not sure if that's ok or not. Also please correct me if I'm wrong somewhere.

Reply:
Okay, keep it as optional, empty string is in 99% of cases a wrong solution. It might be correct in places where we actually want to display some text and the text should be empty.